bokehjs/src/lib/models/tools/math_text.ts

```diff
@@ -0,0 +1,130 @@
+import * as p from "core/properties"
+import {isNumber} from "core/util/types"
+import {Context2d} from "core/util/canvas"
+import {ImageLoader} from "core/util/image"
+import {CanvasImage} from "models/glyphs/image_url"
+import {Model} from "../../model"
+
+
+declare namespace MathJax {
+  function tex2svg(input: string): HTMLElement
+}
+
+export namespace MathText {
+  export type Attrs = p.AttrsOf<Props>
+
+  export type Props = Model.Props & {
+    text: p.Property<string>
+  }
+}
+
+export interface MathText extends MathText.Attrs {}
+
+type Position = {
+  sx: number
+  sy: number
+  x_anchor?: number | "left" | "center" | "right"
+  y_anchor?: number | "top"  | "center" | "baseline" | "bottom"
+}
+
+export class MathText extends Model {
+  override properties: MathText.Props
+
+  image: CanvasImage
+  height: number
+  width: number
+  mathjax_instantiated = false
+
+  constructor(attrs?: Partial<MathText.Attrs>) {
+    super(attrs)
+  }
+
+  static init_MathText(): void {
+    this.define<MathText.Props>(({String}) => ({
+      text: [ String ],
+    }))
+  }
+
+  protected _computed_position(position: Position): {x: number, y: number} {
+    const {width, height} = this
+    const {sx, sy, x_anchor="left", y_anchor="center"} = position
+
+    const x = sx - (() => {
+      if (isNumber(x_anchor))
+        return x_anchor*width
+      else {
+        switch (x_anchor) {
+          case "left": return 0
+          case "center": return 0.5*width
+          case "right": return width
+        }
+      }
+    })()
+
+    const y = sy - (() => {
+      if (isNumber(y_anchor))
+        return y_anchor*height
+      else {
+        switch (y_anchor) {
+          case "top": return 0
+          case "center": return 0.5*height
+          case "bottom": return height
+          case "baseline": return 0.5*height
+        }
+      }
+    })()
+
+    return {x, y}
+  }
+
+  draw_image(ctx: Context2d, position: Position, angle: number): void {
+    if (this.image) {
+      ctx.save()
+      const {sx, sy} = position
+
+      if (angle) {
+        ctx.translate(sx, sy)
+        ctx.rotate(angle)
+        ctx.translate(-sx, -sy)
+      }
+
+      const {x, y} = this._computed_position(position)
+
+      ctx.drawImage(this.image, x, y, this.width, this.height)
+      ctx.restore()
+    }
+  }
+
+  load_image(color: string) {
+    if (!this.mathjax_instantiated) {
+      const script = document.createElement('script')
+      script.src = 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js'
+      script.async = false
+      document.head.appendChild(script)
+      this.mathjax_instantiated = true
+    }
```

Comment:
This will be initialized per model instance. This needs to be changed to be initialized once per bokehjs (not just statically). See `GMapPlotCanvas` for inspiration, but don't follow it exactly as gmaps have very specific loading procedure.

Reply:
Setting `async = false` will kill performance anywhere except in local testing. This needs to be wrapped in a promise and the rest of the code changed appropriately.

Reply:
Moreover, it should be investigated if MathJax could be loaded as a module, to make it impossible for it conflict with a different, possibly incompatible, version already loaded. If loading as module doesn't work, then we may want to try registering it as a unique global. In general, bokehjs should be as transparent to the environment (e.g. a web site it's being embedded into) as possible, and avoid polluting the global namespace, especially when it could conflict with the rest of the environment.

Reply:
I could not make it work as an ESModule (<script type="module">), it is possible to use it as a npm module but it would increase our app size. I changed the validation to resemble your example on `GMapPlotCanvas`. It looks like: 
```js
if (!window._bokeh_mathjax_instantiated && typeof MathJax === 'undefined') {
  const script = document.createElement('script')
  script.src = 'https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js'
  document.head.appendChild(script)

  window._bokeh_mathjax_instantiated = true
}
```

Reply:
> `window._bokeh_mathjax_instantiated = true`

There is no need to burden the global scope with this. Keep this encapsulated in the module that's responsible for loading MathJax.

Reply:
> (...) it is possible to use it as a npm module but it would increase our app size.

We may want to do that as a part of another bokehjs bundle, but the packaging issue is orthogonal to this, so needs to be discussed separately.